app/Controllers/Http/OrdersController.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import Database from '@ioc:Adonis/Lucid/Database';
+import Order from 'App/Models/Order';
+import CreateOrder from 'App/Validators/CreateOrderValidator';
+import UpdateOrder from 'App/Validators/UpdateOrderValidator';
+
+export default class OrdersController {
+    public async index({}: HttpContextContract) {
+        const orders = await Database.from('orders')
+            .join('products', 'orders.product_id', '=', 'products.id')
+            .select('orders.id')
+            .select('orders.quantity')
+            .select('products.name')
+            .select('products.price')
+
+        if (Object.keys(orders).length === 0) {
+            return {
+                status: 'fail',
+                message: 'Failed to retrieve orders'
+            }
+        }
+
+        return {
+            status: 'success',
+            data: orders,
+            message: 'Orders retrieved successfully'
+        }
+    }
+
+    public async store({ request }) {
+        const order = new Order();
+        const payload = await request.validate(CreateOrder);
+
+        order.quantity = payload.quantity;
+        order.productId = payload.productId;
+
+        await order.save();
+
+        if (!order.$isPersisted) {
```

Comment:
So this is the best way to check? The await callback does not return any status or you cannot use try catch to make sure you are fail safe?

Reply:
It is not the best way to check, I am going to clean it up to be better with the try-catch

Reply:
@arnoldkunihira no need to clean up, but just showcase awareenss and a plan and we are good :)